### PR TITLE
Filter Damper class

### LINKED ISSUE #119: Filter Damping addition to the damping module
The Filter Damping method (adapted from [here](https://aip.scitation.org/doi/full/10.1063/1.2728935?casa_token=9v62oae0XN0AAAAA%3AhfV-OcYFBb_uB6cE0yBfsdf9HMIem1vZ0JJVhoJT-WNgxoM5jPXAPfqctpkn3aXP5wN75sVnHLFw)), which basically damps velocities similar to a low pass filter, has shown advantages (reduced noise, oscillations, and larger stable timesteps) in complex scenarios involving Cosserat rods, as noticed by @armantekinalp. Thus, adding this to the damping module (#114) can be an important factor in improving the stability of Cosserat rod simulations.

### PR DESCRIPTION
**Pull request by bhosale2**
Fixes #119, by adding a `FilterDamper` class. @armantekinalp feel free to edit at places (docs, specifically notes) as you see fit.

### CODE REVIEW

**Review by armantekinalp — APPROVED** [review]
LGTM

File: elastica/dissipation.py [hunk 1]
```diff
@@ -151,3 +154,115 @@ def dampen_rates(self, rod: SystemType, time: float):
...
+    integer values (1, 2, etc.) result in aggressive filtering, and can lead to
+    the "physics" being filtered out. While high values (9, 10, etc.) imply
+    minimal filtering, and thus negligible effect on the velocities.
+    Values in the range of 3-7 are usually recommended.
+
+    For details regarding the numerics behind the filtering, refer to:
+
+    Jeanmart, H., & Winckelmans, G. (2007). Investigation of eddy-viscosity
+    models modified using discrete filters: a simplified “regularized variational
+    multiscale model” and an “enhanced field model”. Physics of fluids, 19(5), 055110.
```
Comment (armantekinalp) [inline elastica/dissipation.py:186]: Can we add thesis as reference.

File: elastica/dissipation.py [hunk 2]
```diff
@@ -151,3 +154,115 @@ def dampen_rates(self, rod: SystemType, time: float):
         rod.omega_collection[:] = rod.omega_collection * np.power(
             self.rotational_exponential_damping_coefficient, rod.dilatation
         )
+
+
+class FilterDamper(DamperBase):
```
Comment (skim0119) [inline elastica/dissipation.py:159]: Lets use some more explicit name for the damper.

- `SecondOrderOperator` : Exact name used in the referenced paper.
- `SecondOrderDiscretizedFilter` : Explicit name of what it does intuitively.
- `LaplaceDissipationFilter` : Explicit name for robotics/signal-processing jargon
- <s>`SecondOrderVelocityDissipation`</s>

File: elastica/dissipation.py (hunk 1 above)
Comment (bhosale2) [reply elastica/dissipation.py:186]: The thesis takes the idea from this paper, so is it needed since we already cited the original idea people?

**Review by skim0119 — CHANGES_REQUESTED** [review]

File: elastica/dissipation.py (hunk 1 above)
Comment (armantekinalp) [reply elastica/dissipation.py:186]: Thesis can be the other reference, I think there they present the idea nicely.

File: elastica/dissipation.py [hunk 3]
```diff
@@ -151,3 +154,115 @@ def dampen_rates(self, rod: SystemType, time: float):
...
+    -----
+    The extent of filtering can be controlled by the `filter_order`. Small
+    integer values (1, 2, etc.) result in aggressive filtering, and can lead to
+    the "physics" being filtered out. While high values (9, 10, etc.) imply
+    minimal filtering, and thus negligible effect on the velocities.
+    Values in the range of 3-7 are usually recommended.
+
+    For details regarding the numerics behind the filtering, refer to:
+
+    Jeanmart, H., & Winckelmans, G. (2007). Investigation of eddy-viscosity
```
Comment (armantekinalp) [inline elastica/dissipation.py:184]:
```suggestion
    .. [1] Jeanmart, H., & Winckelmans, G. (2007). Investigation of eddy-viscosity
```

File: elastica/dissipation.py (hunk 1 above)
Comment (armantekinalp) [reply elastica/dissipation.py:186]:
```suggestion
    multiscale model” and an “enhanced field model”. Physics of fluids, 19(5), 055110.
    .. [2] Lorieul, G. (2018). Development and validation of a 2D Vortex Particle-Mesh
     method for incompressible multiphase flows 
     (Doctoral dissertation, Université Catholique de Louvain).
```

File: elastica/dissipation.py [hunk 4]
```diff
@@ -151,3 +154,115 @@ def dampen_rates(self, rod: SystemType, time: float):
...
+    rate_collection : numpy.ndarray
+        2D array containing data with 'float' type.
+        Array containing rod rates (velocities).
+    filter_term: numpy.ndarray
+        2D array containing data with 'float' type.
+        Filter term that modifies rod rates (velocities).
+    filter_order : int
+        Order of the filter.
+
+    Returns
```
[the remark this answers is not in the document]
Comment (bhosale2) [reply elastica/dissipation.py]: 4e606da

File: elastica/dissipation.py [hunk 5]
```diff
@@ -151,3 +154,115 @@ def dampen_rates(self, rod: SystemType, time: float):
...
+    For details regarding the numerics behind the filtering, refer to:
+
+    Jeanmart, H., & Winckelmans, G. (2007). Investigation of eddy-viscosity
+    models modified using discrete filters: a simplified “regularized variational
+    multiscale model” and an “enhanced field model”. Physics of fluids, 19(5), 055110.
+
+    Attributes
+    ----------
+    filter_order : int
+        Order of the filter.
```
[the remark this answers is not in the document]
Comment (bhosale2) [reply elastica/dissipation.py]: 4e606da

**Review by skim0119 — APPROVED** [review]
LGTM. Good work!

**Pull request merged**